sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/nodes/intrinsics/c/LLVMCMathsIntrinsics.java

```diff
@@ -248,6 +248,26 @@ protected LLVM80BitFloat doIntrinsic(LLVM80BitFloat value) {
         }
     }
 
+    @NodeChild(type = LLVMExpressionNode.class)
+    public abstract static class LLVMRound extends LLVMBuiltin {
+
+        @Specialization
+        protected float doIntrinsic(float value) {
+            return (float) Math.round(value);
+        }
+
+        @Specialization
+        protected double doIntrinsic(double value) {
+            return Math.round(value);
+        }
+
+        @Specialization
+        protected LLVM80BitFloat doIntrinsic(LLVM80BitFloat value) {
+            double result = doIntrinsic(value.getDoubleValue());
```

Comment:
I believe this is imprecise

Reply:
Yes, that might lose precision, but in practice it's rarely a problem, since we're rounding anyway. This is only a problem for numbers that are big enough that the precision of `double` is not enough to represent a difference of `1.0` (i.e. numbers bigger than `2^52`).

Is that a problem for your use-case? Then we'd need to do a native call, similar to what we do for regular arithmetic. Unfortunately we don't have any way to directly do 80 bit floating point arithmetic in Java yet.

Reply:
Yep sounds right. Probably not a problem for our use cases! Thanks~